docs/how-to-guides/shared-libraries.rst

```diff
@@ -0,0 +1,200 @@
+
+        This shared library is installed alongside this __init__.py file. Due to
+        lack of rpath support, Windows cannot find shared libraries installed
+        within wheels. So pre-load it.
+        """
+        if os.name == "nt":
+            import ctypes
+            try:
+                from ctypes import WinDLL
+                basedir = os.path.dirname(__file__)
+            except:
+                pass
+            else:
+                dll_path = os.path.join(basedir, "example_lib.dll")
+                if os.path.exists(dll_path):
+                    WinDLL(dll_path)
+
+    _load_sharedlib()
+
+If an internal shared library is not only used as part of a Python package, but
+for example also as a regular shared library in a C/C++ project or as a
+standalone library, then the method shown above won't work - the library has to
+be installed to the default ``libdir`` location. In that case, ``meson-python``
+will detect that the library is going to be installed to ``libdir`` - which is
+not a recommended install location for wheels, and not supported by
+``meson-python``. Instead, ``meson-python`` will do the following *on platforms
+other than Windows*:
+
+1. Install the shared library to ``<project-name>.mesonpy.libs`` (i.e., a
+   top-level directory in the wheel, which on install will end up in
+   ``site-packages``).
+2. Rewrite RPATH entries for install targets that depend on the shared library
+   to point to that new install location instead.
+
+This will make the shared library work automatically, with no other action needed
+from the package author. *However*, currently an error is raised for this situation
+on Windows. This is documented also in :ref:`reference-limitations`.
+
+
+External shared libraries
+=========================
+
+External shared libraries are installed somewhere on the build machine, and
+usually detected by a ``dependency()`` or ``compiler.find_library()`` call in a
+``meson.build`` file. When a Python extension module or executable uses the
+dependency, the shared library will be linked against at build time. On
+platforms other than Windows, an RPATH entry is then added to the built
+extension modulo or executable, which allows the shared library to be loaded at
+runtime.
+
+.. note::
+
+   An RPATH entry alone is not always enough - if the directory that the shared
+   library is located in is not on the loader search path, then it may go
+   missing at runtime. See, e.g., `meson#2121 <https://github.com/mesonbuild/meson/issues/2121>`__
+   and `meson#13046 <https://github.com/mesonbuild/meson/issues/13046>`__ for
+   issues this can cause.
+
+   TODO: describe workarounds, e.g. via ``-Wl,-rpath`` or setting ``LD_LIBRARY_PATH``.
+
+On Windows, the shared library can either be preloaded, or vendored with
+``delvewheel`` in order to make the built Python package usable locally.
+
+
+Publishing wheels which depend on external shared libraries
```

Comment:
There's no mention here of the topic of linking to libraries contained in other wheels. Even if it's not explained in the final draft, it should probably be mentioned and TODO added.

Reply:
Thanks for the suggestion @virtuald. I'm not quite sure I want to add that topic though. Just to clarify, did you mean (a) an actual shared library contained in a wheel (a la `scipy-openblas32`), or (b) a Python package that exports a C API (a la NumPy's C API)?

I think (b) doesn't really fit topic-wise, and (a) isn't recommended, since it requires a bunch of manual workarounds to get to work at all, and then there's no good way to deal with versioning and upgrades.

Reply:
I mean (a), as I think when you start discussing 'external' libraries eventually that thought comes up, so it should be given at least some mention. As I said, even if it's just a footnote that says "you probably shouldn't do this" or something else to that effect, I think that would be appropriate enough for now.

Reply:
Makes sense, thanks - I'll add a note like that.

Reply:
I added this note.